test/tools/jsontests/TransactionTests.cpp

```diff
@@ -38,134 +39,174 @@ namespace fs = boost::filesystem;
+			}
+
+			mObject resultObject;
+			resultObject["sender"] = toString(txFromFields.sender());
+			//resultObject["transaction"] = ImportTest::makeAllFieldsHex(tObj);
+			resultObject["hash"] = toString(txFromFields.sha3());
+			out[test::netIdToString(network)] = resultObject;
+		}
+		catch (Exception const& _e)
+		{
+			//Transaction is InValid
+			cnote << "Transaction Exception: " << diagnostic_information(_e);
+			bool expectInValid = (expectSection["result"].get_str() == "invalid");
+			BOOST_CHECK_MESSAGE(expectInValid, TestOutputHelper::get().testName() + " Check state: Transaction is expected to be valid! (" + test::netIdToString(network) + ")");
+			continue;
+		}
+
+		//Transaction is Valid
+		bool expectValid = (expectSection["result"].get_str() == "valid");
+		BOOST_CHECK_MESSAGE(expectValid, TestOutputHelper::get().testName() + " Check state: Transaction is expected to be invalid! (" + test::netIdToString(network) + ")");
+	}
+	return out;
+}
+
+void TestTransactionTest(json_spirit::mObject const& _o)
+{
+	BOOST_REQUIRE(_o.count("rlp") > 0);
+	string const& testname = TestOutputHelper::get().testName();
+
+	// Theoretical block for transaction check
+	BlockHeader bh;
+	bh.setNumber(1);	//Seal engine below enables network rules from block 0
+	bh.setGasLimit(u256("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
+
+	for (auto const network: test::getNetworks())
+	{
+		Transaction txFromRlp;
+		string networkname = test::netIdToString(network);
+		BOOST_REQUIRE_MESSAGE(_o.count(networkname) > 0, testname + " Transaction test missing network results! (" + networkname + ")");
+		BOOST_REQUIRE(_o.at(networkname).type() == json_spirit::obj_type);
+		mObject obj = _o.at(networkname).get_obj();
+		try
+		{
+			bytes stream = importByteArray(_o.at("rlp").get_str());
+			RLP rlp(stream);
+
+			txFromRlp = Transaction(rlp.data(), CheckTransaction::Everything);
+			ChainParams params(genesisInfo(network));
+
+			// Test networks has forkblocks set to 0 if rules are enabled
+			bool onConstantinople = (bh.number() >= params.constantinopleForkBlock);
+			bool onConstantinopleAndZeroSig = onConstantinople && txFromRlp.hasZeroSignature();
+			unique_ptr<SealEngineFace> se(params.createSealEngine());
+
+			se->verifyTransaction(ImportRequirements::Everything, txFromRlp, bh, 0);
+			if (!(txFromRlp.signature().isValid() || onConstantinopleAndZeroSig))
+				BOOST_THROW_EXCEPTION(Exception() << errinfo_comment(testname + "transaction from RLP signature is invalid (" + networkname + ")") );
+		}
+		catch (Exception const& _e)
+		{
+			cnote << testname;
+			cnote << "Transaction Exception: " << diagnostic_information(_e);
+			BOOST_CHECK_MESSAGE(obj.count("hash") == 0,
+				testname + "A transaction object should not be defined because the RLP is invalid! (" + networkname + ")");
+			continue;
+		}
+		catch (...)
```

Comment:
Are we really expecting this arbitrary exceptions happening from invalid RLP?

Reply:
this one is to be safe.

Reply:
Let's remove and see what happens.